### ISSUE TITLE
!! NOTICE: "ValidationError: Circular dependency between resources" for SecretsManager RotationSchedule with rotationLambda

### ISSUE THREAD

**Issue by skinny85** [aws/aws-cdk#14868]
Please add your +1 👍 to let us know you have encountered this
---

### Overview:

When using a `RotationSchedule` with a `rotationLambda`, there is a circular dependency that happens between the default IAM Policy of the `rotationLambda` Role, and the Function itself:

```ts
import { Construct, Stack, StackProps } from '@aws-cdk/core';
import * as lambda from '@aws-cdk/aws-lambda';
import * as secretsmanager from '@aws-cdk/aws-secretsmanager';

export class TestStack extends Stack {
    constructor(scope: Construct, id: string, props?: StackProps) {
        super(scope, id, props);

        const secret = new secretsmanager.Secret(this, 'Secret');
        const rotationLambda = new lambda.Function(this, 'RotateLambda', {
            runtime: lambda.Runtime.NODEJS_10_X,
            code: lambda.Code.fromInline('export.handler = event => event;'),
            handler: 'index.handler',
        });
        new secretsmanager.RotationSchedule(this, 'RotationSchedule', {
            secret,
            rotationLambda: rotationLambda,
        });
    }
}
```

#### Complete Error Message:

```
$ yarn cdk deploy
yarn run v1.22.10
TestStack: deploying...
TestStack: creating CloudFormation changeset...

 ❌  TestStack failed: ValidationError: Circular dependency between resources: [RotateLambdaCCE05B95, RotationSchedule2A725270, RotateLambdaServiceRoleDefaultPolicy12636DDF, RotateLambdaInvokeServicePrincipalsecretsmanageramazonawscom32ADD300]
Circular dependency between resources: [RotateLambdaCCE05B95, RotationSchedule2A725270, RotateLambdaServiceRoleDefaultPolicy12636DDF, RotateLambdaInvokeServicePrincipalsecretsmanageramazonawscom32ADD300]
```

### Workaround:

This regression was introduced in CDK version [`1.104.0`](https://github.com/aws/aws-cdk/releases/tag/v1.104.0), so if you use a version of CDK `1.103.0` or earlier, you will not be affected.

### Related Issues:

This was added in #14471.

**Comment by skinny85** [comment]
Revert of the faulty PR: https://github.com/aws/aws-cdk/pull/14869

**Comment by skinny85** [comment]
PR with the patch release: https://github.com/aws/aws-cdk/pull/14881

**Comment by skinny85** [comment]
The fix has been released in [CDK version `1.106.1`](https://github.com/aws/aws-cdk/releases/tag/v1.106.1). Resolving.
